docs/platforms/react-native/migration/v5-to-v6.mdx

```diff
@@ -0,0 +1,260 @@
++ const reactNavigationIntegration = new Sentry.reactNavigationIntegration();
+
+Sentry.init({
+  tracesSampleRate: 1.0,
+  integrations: [
+-    new Sentry.ReactNativeTracing({ routingInstrumentation }),
++    reactNavigationIntegration,
+  ],
+});
+```
+
+```javascript {tabTitle:React Native Navigation} diff
+import Sentry from '@sentry/react-native';
+import { Navigation } from 'react-native-navigation';
+
+Sentry.init({
+  tracesSampleRate: 1.0,
+  integrations: [
+-    new Sentry.ReactNativeTracing({
+-      routingInstrumentation: new Sentry.ReactNativeNavigationInstrumentation(
+-        Navigation,
+-      ),
+-    }),
++    new Sentry.reactNativeNavigationIntegration({
++      navigation: Navigation,
++    }),
+  ],
+});
+```
+
+### React Navigation version 4 is not supported
+
+`ReactNavigationV4Instrumentation` was removed in the SDK version 6. If you are using React Navigation version 4, you need to upgrade to version 5 or never.
+
+### RoutingInstrumentation for Custom Navigation was removed
+
+`RoutingInstrumentation` was removed in the SDK version 6. If you are using custom navigation, use `startIdleNavigationSpan` fucntion.
+
+```javascript {tabTitle:Direct} diff
+- const routingInstrumentation = new Sentry.RoutingInstrumentation();
+
+Sentry.init({
+  tracesSampleRate: 1.0,
+  integrations: [
+-    new Sentry.ReactNativeTracing({
+-      routingInstrumentation,
+-    }),
+  ],
+})
+
+const App = () => {
+  <SomeNavigationLibrary
+    onRouteWillChange={(newRoute) => {
+-      routingInstrumentation.onRouteWillChange({
++      Sentry.startIdleNavigationSpan({
+        name: newRoute.name,
+        op: 'navigation'
+      });
+    }}
+  />
+};
+```
+
+```javascript {tabTitle:Integration} diff
+// Before
+import Sentry from '@sentry/react-native';
+
+-class CustomInstrumentation extends Sentry.RoutingInstrumentation {
+-  constructor(navigator) {
+-    super();
+-
+-    this.navigator.registerRouteChangeListener(
+-      this.routeListener.bind(this),
+-    );
+-  }
+-
+-  routeListener(newRoute) {
+-    this.onRouteWillChange({
++const customNavigationIntegration = ({navigator}) => {
++  navigator.registerRouteChangeListener((newRoute) => {
++    Sentry.startIdleNavigationSpan({
+      name: newRoute.name,
+      op: "navigation",
+-    });
+-  }
+-}
++    })
++  });
++
++  return {
++    name: "CustomNavigation",
++  };
++};
+
+Sentry.init({
+  tracesSampleRate: 1.0,
+  integrations: [
+-    new Sentry.ReactNativeTracing({
+-      routingInstrumentation: new CustomInstrumentation(navigator),
+-    }),
++    customNavigationIntegration({navigator}),
+  ],
+});
+```
+
+### `beforeNavigate` replaced by `beforeStartSpan`
+
+The `beforeNavigate` option was removed in the SDK version 6. Use the `beforeStartSpan` option instead. The `beforeStartSpan` option is a function that is called before starting a navigation span. This function can't stop the span from being started, but it can modify the span start options before it starts.
```

Comment:
```suggestion

The `beforeNavigate` option was removed in version 6 of the SDK . Use the `beforeStartSpan` option instead. The `beforeStartSpan` option is a function that's called before starting a navigation span. This function can't stop the span from being started, but it can modify the span start options before it starts.
```